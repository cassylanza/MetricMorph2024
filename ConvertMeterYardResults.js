import React, { Component } from 'react'
import { TouchableOpacity, Text, View, StyleSheet, ScrollView } from 'react-native'
import { Actions } from 'react-native-router-flux'


const ConvertMeterYardResults = () => {

  const goToTrackMenu = () => {
    Actions.trackmenu1()
  }

  const goToSwimMenu = () => {
    Actions.swimmenu1()
  }

return (
  <View>
    <Text style = {styles.mainResult}> MM:SS:TH </Text>


    <View style = {styles.containerTaskBar}>
      <TouchableOpacity onPress = {goToSwimMenu}>
        <Text style = {styles.textTaskBarSwim}> Swim Menu </Text>
      </TouchableOpacity>
      <TouchableOpacity onPress = {goToTrackMenu}>
        <Text style = {styles.textTaskBar}> Track Menu </Text>
      </TouchableOpacity>
    </View>
  </View>
)



}
export default ConvertMeterYardResults

const styles = StyleSheet.create ({
  container: {
      padding: 10,
      marginTop: 3,
      //backgroundColor: '#d9f9b1',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'flex-end',
      borderWidth: 1
   },
   text: {
      color: '#4f603c'
   },
   mainResult: {
     width: 250,
     height: 75,
     textAlign: 'center',
     borderWidth: 1,
     padding: 18,
     fontSize: 30,
     borderColor: 'black',
     backgroundColor: 'white',
     marginTop: 200,
     marginBottom: 385,
     marginLeft: 60
   },
   containerTaskBar: {
     flexDirection: 'row',
     justifyContent: 'space-between',
     alignItems: 'center',
     height: 20
   },
   textTaskBar: {
     width: 188,
     height: 100,
     textAlign: 'center',
     borderWidth: 1,
     padding: 37,
     fontSize: 20,
     borderColor: 'black',
     backgroundColor: 'white'
   },
   textTaskBarSwim: {
     width: 188,
     height: 100,
     textAlign: 'center',
     borderWidth: 1,
     padding: 37,
     fontSize: 20,
     borderColor: 'black',
     backgroundColor: 'white',
     fontWeight: 'bold',
     color: 'blue'
   }
})
