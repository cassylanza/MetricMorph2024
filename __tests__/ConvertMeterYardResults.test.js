import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import { NavigationContainer } from '@react-navigation/native';
import { Actions } from 'react-native-router-flux'; // Mocking react-native-router-flux
import '@testing-library/jest-dom';
import ConvertMeterYardResults from '../ConvertMeterYardResults.js';

jest.mock('react-native-router-flux', () => ({
  Actions: {
    swimmenu1: jest.fn(),
    trackmenu1: jest.fn(),
    convperfinfoentry: jest.fn(),
    conversionresults: jest.fn(),
    splitsinfoentry: jest.fn(),
    splitsresults: jest.fn(),
    effortinfoentry: jest.fn(),
    effortresults: jest.fn(),
  },
}));

describe('ConvertMeterYardResults', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ConvertMeterYardResults />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('result time shows', () => {
    const { getByText } = render(<ConvertMeterYardResults />);
    expect(getByText("MM:SS:TH")).toBeTruthy();
  });



  test('show swim menu when swim menu button pressed', () => {
    render(
      <NavigationContainer>
        <ConvertMeterYardResults />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText('Swim Menu'));

    expect(screen.getByText('Swim Menu')).toBeTruthy();
  });

  test('show track menu when track menu button pressed', () => {
    render(
      <NavigationContainer>
        <ConvertMeterYardResults />
      </NavigationContainer>
    )

    fireEvent.press(screen.getByText('Track Menu'));

    expect(screen.getByText('Track Menu')).toBeTruthy();
  });




})
