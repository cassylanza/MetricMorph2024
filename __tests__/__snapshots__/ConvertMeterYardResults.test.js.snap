// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConvertMeterYardResults renders correctly 1`] = `
<View>
  <Text
    style={
      {
        "backgroundColor": "white",
        "borderColor": "black",
        "borderWidth": 1,
        "fontSize": 30,
        "height": 75,
        "marginBottom": 385,
        "marginLeft": 60,
        "marginTop": 200,
        "padding": 18,
        "textAlign": "center",
        "width": 250,
      }
    }
  >
     MM:SS:TH 
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 20,
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderWidth": 1,
            "color": "blue",
            "fontSize": 20,
            "fontWeight": "bold",
            "height": 100,
            "padding": 37,
            "textAlign": "center",
            "width": 188,
          }
        }
      >
         Swim Menu 
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "backgroundColor": "white",
            "borderColor": "black",
            "borderWidth": 1,
            "fontSize": 20,
            "height": 100,
            "padding": 37,
            "textAlign": "center",
            "width": 188,
          }
        }
      >
         Track Menu 
      </Text>
    </View>
  </View>
</View>
`;
